revel/request_extractor.go

```diff
@@ -0,0 +1,91 @@
+package bugsnagrevel
+
+import (
+	"strings"
+
+	bugsnag "github.com/bugsnag/bugsnag-go"
+	"github.com/revel/revel"
+)
+
+type headerExposer interface {
+	GetAll(string) []string
+}
+
+func extractRequestData(req *revel.Request, paramsFilters []string) *bugsnag.RequestJSON {
+	return &bugsnag.RequestJSON{
+		ClientIP: req.RemoteAddr,
+		// Revel requires you to know the name of the headers up front, which
+		// makes this bit tricky, especially if the application uses custom
+		// headers
+		Headers:    extractHeaders(req.Header, paramsFilters),
+		HTTPMethod: req.Method,
+		URL:        req.URL.String(),
+		Referer:    req.Header.Get("referer"),
+	}
+}
+
+// extractHeader is a best-guess workaround for pulling out as many headers as possible in a Revel request.
+// Extracts all the standard HTTP headers, and some non-standard, but common headers.
+func extractHeaders(h headerExposer, paramsFilters []string) map[string]string {
+	m := make(map[string]string)
+	// Standard headers
+	addHeader(paramsFilters, m, h, "Accept")
```

Comment:
Would it be possible to add the header keys to some sort of collection then iterating over it, to avoid multiple calls to `addHeader`?